pkg/sync/http/http_sync_test.go

```diff
@@ -55,6 +56,37 @@ func TestSimpleSync(t *testing.T) {
 	}
 }
 
+func TestFailedSync(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	resp := "not returned"
+
+	mockCron := syncmock.NewMockCron(ctrl)
+	mockCron.EXPECT().AddFunc(gomock.Any(), gomock.Any()).DoAndReturn(func(spec string, cmd func()) error {
+		return errors.New("fake")
+	})
+
+	mockClient := syncmock.NewMockClient(ctrl)
+	mockClient.EXPECT().Do(gomock.Any()).Return(&http.Response{Body: io.NopCloser(strings.NewReader(resp))}, nil)
+
+	httpSync := Sync{
+		URI:         "http://localhost",
+		Client:      mockClient,
+		Cron:        mockCron,
+		BearerToken: "",
+		LastBodySHA: "",
+		Logger:      logger.NewLogger(nil, false),
+	}
+
+	ctx := context.Background()
+	dataSyncChan := make(chan sync.DataSync)
+
+	// this should return immediately.
+	err := httpSync.Sync(ctx, dataSyncChan)
+	if err == nil {
+		t.Errorf("expected error")
```

Comment:
I assume the error is returned due to invalid cron expression ? If so, can we improve the test log 🤔

Reply:
Resolved by completely removing this check...we know the expression is valid since we have it hard coded.